pkg/minishift/config/validations.go

```diff
@@ -187,3 +187,35 @@ func IsValidNetmask(name string, mask string) error {
 
 	return nil
 }
+
+func IsValidPort(name string, p string) error {
+	port, err := strconv.Atoi(p)
+	if err != nil {
+		return fmt.Errorf("%s:%v", name, err)
+	}
+	if numInRange(port, 1024, 65536) {
+		if err := isPortAccessible(port); err != nil {
+			return fmt.Errorf("Port %d is not accessible: %v", port, err)
+		}
+		return nil
+	} else {
+		return fmt.Errorf("Port %d is out of range use a port in the range (1024-65536)", port)
+	}
+}
+
+func numInRange(num int, start int, end int) bool {
+	if num >= start && num <= end {
+		return true
+	}
+	return false
+}
+
+func isPortAccessible(port int) error {
```

Comment:
The first thing when I see this type of function `isPortAccessible`, I except bool to be returned.
Like https://github.com/minishift/minishift/blob/master/pkg/util/utils.go#L187

Please make the changes.

Reply:
Yeah that is true, but i made that choice because i wanted to print the error, if it failed so needed to return it to the upper level. But the error is of only one type here, which is bind error so, I am okay with returning a bool. Making the change. :)